frontend/src/components/Forms/LoginPage/LoginPage.tsx

```diff
@@ -0,0 +1,125 @@
+import { useRef, useState } from "react";
+import classes from "../Forms.module.scss";
+
+function LoginPage({ onClick }: { onClick: () => void }) {
+  const [formData, setFormData] = useState({
+    username: "",
+    password: "",
+  });
+  const [usernameFound, setUsernameFound] = useState(true);
+  const [passwordMatch, setPasswordMatch] = useState(true);
+  const [usernameEmpty, setUsernameEmpty] = useState(false);
+  const [passwordEmpty, setPasswordEmpty] = useState(false);
+  const [isProcessing, setIsProcessing] = useState(false);
+  const usernameField = useRef<HTMLInputElement>(null);
+  const passwordField = useRef<HTMLInputElement>(null);
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setFormData({
+      ...formData,
+      [e.target.name]: e.target.value,
+    });
+    if (!usernameFound) setUsernameFound(true);
+    if (!passwordMatch) setPasswordMatch(true);
+    if (usernameEmpty) setUsernameEmpty(false);
+    if (passwordEmpty) setPasswordEmpty(false);
+    usernameField.current?.classList.remove(classes.inputError);
+    passwordField.current?.classList.remove(classes.inputError);
+  };
+
+  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    if (isProcessing) return;
+    setIsProcessing(true);
+
+    if (formData.username === "") {
```

Comment:
change to match what we were talking abt with outsourcing these checks to another fn